docs/site/Interceptors.md

```diff
@@ -0,0 +1,552 @@
+---
+lang: en
+title: 'Interceptors'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Interceptors.html
+---
+
+## Overview
+
+Interceptors are reusable functions to provide aspect-oriented logic around
+method invocations. There are many use cases for interceptors, such as:
+
+- Add extra logic before / after method invocation, for example, logging or
+  measuring method invocations.
+- Validate/transform arguments
+- Validate/transform return values
+- Catch/transform errors, for example, normalize error objects
+- Override the method invocation, for example, return from cache
+
+## Basic use
+
+### Interceptors on controllers
+
+Interceptors are supported for public controller methods (including both static
+and prototype) and handler functions for REST routes.
+
+Controller methods decorated with `@intercept` are invoked with applied
+interceptors for corresponding routes upon API requests.
+
+```ts
+import {intercept} from '@loopback/context';
+
+@intercept(log)
+export class OrderController {
+  @intercept('caching-interceptor')
+  async listOrders(userId: string) {
+    // ...
+  }
+}
+```
+
+It's also possible to configure global interceptors that are invoked before
+method level interceptors. For example, the following code registers a global
+`caching-interceptor` for all methods.
+
+```ts
+app
+  .bind('caching-interceptor')
+  .toProvider(CachingInterceptorProvider)
+  .apply(asInterceptor);
+```
+
+Global interceptors are also executed for handler functions without a controller
+class:
+
+```ts
+/**
+ * OpenAPI operation spec for `toLowerCase(text: string)`
+ */
+const toLowerCaseOperationSpec = anOperationSpec()
+  .withOperationName('toLowerCase')
+  .withParameter({
+    name: 'text',
+    in: 'path',
+    schema: {
+      type: 'string',
+    },
+  })
+  .withStringResponse()
+  .build();
+
+/**
+ * A plain function to convert `text` to lower case
+ * @param text
+ */
+function toLowerCase(text: string) {
+  return text.toLowerCase();
+}
+
+app.route('get', '/toLowerCase/{text}', toLowerCaseOperationSpec, toLowerCase);
```

Comment:
Can we use a simpler example please? IIUC, you want to demonstrate that interceptors are invoked for handler functions too. This code contains lot of low-level details related to route definition but unrelated to interceptors. 

I think it's best to not duplicate content we have elsewhere and simply point readers to https://loopback.io/doc/en/lb4/Routes.html#using-partial-openapi-spec-fragments where handler functions are explained in more details.